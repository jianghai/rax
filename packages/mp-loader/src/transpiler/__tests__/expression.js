const { hasExpression, transformExpression } = require('../expression');

describe('expression', () => {
  describe('hasExpression', () => {
    it('no expression', () => {
      expect(hasExpression('exp'))
        .toBeFalsy();
    });

    it('simple expression', () => {
      expect(hasExpression('{{exp}}'))
        .toBeTruthy();
    });

    it('complex expression', () => {
      expect(hasExpression('foo {{bar}}'))
        .toBeTruthy();
      expect(hasExpression('foo {{ bar + {} }}'))
        .toBeTruthy();
    });
  });

  describe('transformExpression', () => {
    it('no expression', () => {
      expect(transformExpression('exp')).toEqual('("exp")');
    });

    it('simple expression', () => {
      expect(transformExpression('{{exp}}')).toEqual('(exp)');
    });

    it('complex expression', () => {
      expect(transformExpression('hello {{exp}}')).toEqual('("hello ") + (exp)');

      expect(
        transformExpression('{{[{},{}]}}')
      ).toEqual('([{}, {}])');

      expect(
        transformExpression('{{x:{y:1}}}')
      ).toMatchSnapshot();

      expect(
        transformExpression('{{x}} {{y}}')
      ).toEqual('(x) + (" ") + (y)');

      expect(
        transformExpression(`
           {{
              a: 1
           }}
        `)
      ).toMatchSnapshot();
    });
  });
});
