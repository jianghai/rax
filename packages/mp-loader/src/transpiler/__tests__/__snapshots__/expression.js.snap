// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expression transformExpression complex expression 1`] = `
"(({
  x: {
    y: 1
  }
}))"
`;

exports[`expression transformExpression complex expression 2`] = `
"(({
  a: 1
}))"
`;
